src/transactions/transactionBytes.js

```diff
@@ -12,181 +12,126 @@
 
+const byteSizes = {
+	TYPE: 1,
+	TIMESTAMP: 4,
+	MULTISIGNATURE_PUBLICKEY: 32,
+	RECIPIENT_ID: 8,
+	AMOUNT: 8,
+	SIGNATURE_TRANSACTION: 64,
+	SECOND_SIGNATURE_TRANSACTION: 64,
+	DATA: 64,
+};
+
 /**
- * @method getEmptyBytesForTransaction
- * @param transaction Object
- * @return {object}
+ * @method getAssetBytesHelper
+ * @return {Buffer}
  */
 
-function getEmptyBytesForTransaction(transaction) {
+function getAssetBytesHelper(transaction) {
 	/**
 	 * @method isSendTransaction
-	 * @return {object}
+	 * @return {Buffer}
 	 */
 
 	function isSendTransaction() {
-		return {
-			assetBytes: null,
-			assetSize: 0,
-		};
+		return transaction.asset.data
+			? Buffer.from(transaction.asset.data, 'utf8')
+			: Buffer.alloc(0);
 	}
 
 	/**
 	 * @method isSignatureTransaction
-	 * @return {object}
+	 * @return {Buffer}
 	 */
 
 	function isSignatureTransaction() {
-		const bb = new ByteBuffer(32, true);
-		const publicKey = transaction.asset.signature.publicKey;
-		const publicKeyBuffer = Buffer.from(publicKey, 'hex');
-
-		for (let i = 0; i < publicKeyBuffer.length; i++) {
-			bb.writeByte(publicKeyBuffer[i]);
-		}
-
-		bb.flip();
-		const signatureBytes = new Uint8Array(bb.toArrayBuffer());
-
-		return {
-			assetBytes: signatureBytes,
-			assetSize: 32,
-		};
+		return Buffer.from(transaction.asset.signature.publicKey, 'hex');
 	}
 
 	/**
 	 * @method isDelegateTransaction
-	 * @return {object}
+	 * @return {Buffer}
 	 */
 
 	function isDelegateTransaction() {
-		return {
-			assetBytes: Buffer.from(transaction.asset.delegate.username),
-			assetSize: Buffer.from(transaction.asset.delegate.username).length,
-		};
+		return Buffer.from(transaction.asset.delegate.username, 'utf8');
 	}
 
 	/**
 	 * @method isVoteTransaction
-	 * @return {object}
+	 * @return {Buffer}
 	 */
 
 	function isVoteTransaction() {
-		const voteTransactionBytes = (Buffer.from(transaction.asset.votes.join('')) || null);
-
-		return {
-			assetBytes: voteTransactionBytes,
-			assetSize: (voteTransactionBytes.length || 0),
-		};
+		return Buffer.from(transaction.asset.votes.join(''));
 	}
 
 	/**
 	 * @method isMultisignatureTransaction
-	 * @return {object}
+	 * @return {Buffer}
 	 */
 
 	function isMultisignatureTransaction() {
-		const MINSIGNATURES = 1;
-		const LIFETIME = 1;
-		const keysgroupBuffer = Buffer.from(transaction.asset.multisignature.keysgroup.join(''), 'utf8');
-
-		const bb = new ByteBuffer(MINSIGNATURES + LIFETIME + keysgroupBuffer.length, true);
-		bb.writeByte(transaction.asset.multisignature.min);
-		bb.writeByte(transaction.asset.multisignature.lifetime);
-		for (let i = 0; i < keysgroupBuffer.length; i++) {
-			bb.writeByte(keysgroupBuffer[i]);
-		}
-		bb.flip();
-
-		bb.toBuffer();
-		const multiSigBuffer = new Uint8Array(bb.toArrayBuffer());
-
-		return {
-			assetBytes: multiSigBuffer,
-			assetSize: multiSigBuffer.length,
-		};
+		const multisigTransactionAsset = transaction.asset.multisignature;
+		const minBuffer = Buffer.alloc(1);
```

Comment:
Prefer allocating directly when creating the Buffer: `Buffer.alloc(1, multisigTransactionAsset.min)`